referenceframe/frame_system.go

```diff
@@ -308,6 +309,48 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD in a framesystem and returns a map where all elements are the point representation
+// of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	for _, name := range system.FrameNames() {
+		currentFrame := system.Frame(name)
+		parent, err := system.Parent(currentFrame)
+		if err != nil {
```

Comment:
won't this cause errors when it finds the World frame which has no parent?

Reply:
youre right, it will cause errors. To take care of this I now do:
```
	if system.FrameNames() == nil {
		return vectorMap, errors.New("cannot convert empty frame system")
	}
```

Reply:
I think he actually means you could work your way through this range loop in an otherwise valid frame system where the `World` frame not having a parent will unceremoniously dump you out.

Instead it needs to have something to check for the World and handle that appropriately. A check for `FrameNames == nil` seems like a good thing to have _in addition_ to what Ray is describing.

Reply:
> A check for FrameNames == nil seems like a good thing to have in addition to what Ray is describing.

It's not clear to me that we should be erroring on an empty framesystem, rather than simply returning an empty array of points. The UI would continuously throw/have to handle errors in that case for a robot until something is added to the frame system.

Reply:
@raybjork @spieswl if this would fail on World, then the current tests wouldn't pass.

`World` is not itself a member of the collection of frames in a frame system, but rather is held separately, so calling something like `FrameNames` will not ever hit `World` itself.

Reply:
> The UI would continuously throw/have to handle errors in that case for a robot until something is added to the frame system.

ok, in this case, I will remove the `FrameNames == nil` check entirely.